In [44]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

data = pd.read_csv('MNIST_dataset/train.csv')

data = np.array(data)
x = data[:, 1:].T
y = data[:, 0].T
print(f"x_shape = {x.shape}")
print(f"y_shape = {y.shape}")

x_shape = (784, 42000)
y_shape = (42000,)


Vamos a crear una red neuronal con la siguiente estructura: (784 x 12 x 12 x 10)

![waterfall](nn.png)

Cambiamos el output deseado de número a vector con 1 en el correcto y 0 en el resto

In [45]:
def one_hot_encode(number, num_classes = 10):
    # Crear un vector de ceros de longitud igual al número de clases
    one_hot = np.zeros(num_classes)
    # Colocar un 1 en la posición correspondiente al número
    one_hot[number] = 1
    return one_hot
y_one_hot = np.array([one_hot_encode(num) for num in y])
y_one_hot

array([[0., 1., 0., ..., 0., 0., 0.],
       [1., 0., 0., ..., 0., 0., 0.],
       [0., 1., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 1., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 1.]])